service/worker/deployment/deployment_workflow.go

```diff
@@ -0,0 +1,150 @@
+		logger               sdklog.Logger
+		metrics              sdkclient.MetricsHandler
+		deploymentlocalState *DeploymentLocalState
+	}
+)
+
+var (
+	defaultActivityOptions = workflow.ActivityOptions{
+		ScheduleToCloseTimeout: 1 * time.Hour,
+		StartToCloseTimeout:    30 * time.Second,
+		RetryPolicy: &temporal.RetryPolicy{
+			InitialInterval: 1 * time.Second,
+			MaximumInterval: 60 * time.Second,
+		},
+	}
+)
+
+const (
+	UpdateDeploymentSignalName        = "update_deployment"
+	UpdateDeploymentBuildIDSignalName = "update_deployment_build_id"
+
+	DeploymentWorkflowIDPrefix = "temporal-sys-deployment:"
+)
+
+func DeploymentWorkflow(ctx workflow.Context, deploymentWorkflowArgs DeploymentWorkflowArgs) error {
+	deploymentWorkflowRunner := &DeploymentWorkflowRunner{
+		deploymentlocalState: &DeploymentLocalState{
+			NamespaceName:  deploymentWorkflowArgs.NamespaceName,
+			NamespaceID:    deploymentWorkflowArgs.NamespaceID,
+			DeploymentName: "", // TODO Shivam - extract the Deployment
+			BuildID:        "", // TODO Shivam - extract BuildID from the workflowID
+			TaskQueues:     deploymentWorkflowArgs.TaskQueues,
+		},
+		ctx:     ctx,
+		logger:  sdklog.With(workflow.GetLogger(ctx), "wf-namespace", deploymentWorkflowArgs.NamespaceName),
+		metrics: workflow.GetMetricsHandler(ctx).WithTags(map[string]string{"namespace": deploymentWorkflowArgs.NamespaceName}),
+	}
+	return deploymentWorkflowRunner.run()
+}
+
+func (d *DeploymentWorkflowRunner) run() error {
+
+	// Set up Query Handlers here:
+
+	// Fetch signal channels
+	updateDeploymentSignalChannel := workflow.GetSignalChannel(d.ctx, UpdateDeploymentSignalName)
+	updateBuildIDSignalChannel := workflow.GetSignalChannel(d.ctx, UpdateDeploymentBuildIDSignalName)
+
+	selector := workflow.NewSelector(d.ctx)
+	selector.AddReceive(updateDeploymentSignalChannel, func(c workflow.ReceiveChannel, more bool) {
+		// Process Signal
+
+	})
+	selector.AddReceive(updateBuildIDSignalChannel, func(c workflow.ReceiveChannel, more bool) {
+		// Process Signal
+	})
+
+	// async draining before CAN
+	for !workflow.GetInfo(d.ctx).GetContinueAsNewSuggested() || selector.HasPending() {
+		selector.Select(d.ctx)
+	}
+
+	/*
+
+		 	Posting this as a reminder to limit the number of signals coming through since we use CAN:
+
+			Workflows cannot have infinitely-sized history and when the event count grows too large, `ContinueAsNew` can be returned
+			to start a new one atomically. However, in order not to lose any data, signals must be drained and any other futures
+			that need to be reacted to must be completed first. This means there must be a period where there are no signals to
+			drain and no futures to wait on. If signals come in faster than processed or futures wait so long there is no idle
+			period, `ContinueAsNew` will not happen in a timely manner and history will grow.
+
+	*/
+
+	d.logger.Debug("Deployment doing continue-as-new")
+	workflowArgs := DeploymentWorkflowArgs{
```

Comment:
I think you want to just pass `d.deploymentLocalState` as a whole to CAN?

Reply:
Yeah, you got that right. However, CAN would require the workflow arguments to be of the expected type (`DeploymentWorkflowArgs`) and hence made the structure as shown.